projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/AuthenticatedEmailNotification.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.notification;
+
+import java.util.Arrays;
+import java.util.Properties;
+import javax.mail.Address;
+import javax.mail.Authenticator;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.SendFailedException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class AuthenticatedEmailNotification {
+
+  private EmailPropertiesConfiguration emailPropertiesConfiguration;
+  private final Session session;
+  private static final String CONTENT_TYPE = "text/html; charset=utf-8";
+
+  public AuthenticatedEmailNotification(EmailPropertiesConfiguration emailPropertiesConfiguration) {
+    this.emailPropertiesConfiguration = emailPropertiesConfiguration;
+    Properties properties = new Properties();
+    properties.setProperty("mail.transport.protocol", "smtp");
+    properties.putAll(this.emailPropertiesConfiguration.smtpWithPrefix());
+    session =
+        Session.getInstance(
+            properties,
+            getAuthenticator(
+                emailPropertiesConfiguration.getUsername(),
+                emailPropertiesConfiguration.getPassword()));
+  }
+
+  public void send(NotificationContent notificationContent) throws MessagingException {
```

Comment:
I realize it's a draft. 
So I hope in latest version we are going to remove the duplication. 

But since Iam looking into it, let me share some ideas. 

We can either create one class and handle autneticaion internally n the class or we can still have 2 classes and employ decorator pattern.